hw/ip/csrng/doc/_index.md

```diff
@@ -91,3 +91,392 @@ This document aims to make the distinction between physical entropy and factory
+    <td>flags</td>
+    <td> Command Flags: Specific flags associated with a command. Used to allow additional features per command. Flags are defined as flag0, flag1, flag2, and flag3, where flag0 is bit 8, and flag1 is bit 9, etc. Flag0 is used for the `instantiate` command. All others are reserved.
+    </td>
+  </tr>
+  <tr>
+    <td>30:12</td>
+    <td>glen</td>
+    <td> Generate Length: Only defined for the generate command, this field is the total number of crytographic entropy bits
+         requested. The NIST reference name is `max_number_of_bit_per_request`, and this field size supports the maximum size allowed.
+         Each unit represents 128 bits of entropy returned. For example, a value of 8 would return a total of 1024 bits.
+         The maximum value for this field is 2^19.
+    </td>
+  </tr>
+  <tr>
+    <td>31</td>
+    <td>resv</td>
+    <td> Unused and reserved.
+    </td>
+  </tr>
+</table>
+
+#### Command Description                                
+The command field of the application command header is described in detail in the table below. 
+Actions as to what the commands will perform, along with what flags are supported, are also described in this table.
+        
+
+<table>
+<caption>Application Interface Command Description</caption>
+<thead>
+  <tr>
+    <th>Command Name</th>
+    <th>Encoded Value</th>
+    <th>Description</th>
+  </tr>
+</thead>
+<tbody>
+  <tr>
+    <td>Instantiate</td>
+    <td>0x1</td>
+    <td> Initializes an instance in CSRNG. For seeding, if `flag0` is not set and `clen` is zero, then a seed of 
+         of only the entropy source seed will be used. If `flag0` is not set and `clen` is non-zero, then the seed will 
+         xor'ed with the provided additional data.  If `flag0` is set and `clen` is zero, then a seed of zero with no
+         entropy source seed will be used. If `flag0` is set and `clen` is non-zero, then only the provided additional
+         data will be used as the seed. When the `instantiate` command is completed, the active bit in the CSRNG
+         working state will be set.
+    </td>
+  </tr>
+  <tr>
+    <td>Reseed</td>
+    <td>0x2</td>
+    <td> Reseeds an existing instance in CSRNG. When `clen` is set to zero for this command, only the entropy
+         source seed will be used to reseed the instance. if `clen` is set to non-zero (up to twelve), the additional
+         data provided with be xor'ed with the entropy source seed.
+    </td>
+  </tr>
+  <tr>
+    <td>Generate</td>
+    <td>0x3</td>
+    <td> Starts a request to CSRNG to generate crytographic entropy bits. The `glen` field represents how many 128-bit
+         words are to be returned to the application interface. The `glen` field needs to be a minimum value of one.
+         The NIST reference to the `prediction_resistance_flag` is not directly supported as a flag. It is the 
+         resposibility of the application interface to reseed before the generate command to properly support
```

Comment:
strictly speaking, it's the thing (hardware block or software) on the other end of the application interface that is responsible for reseeding, right?  said another way, the application interface logic doesn't do this autonomously, or at least I don't expect it to.

Reply:
Yes, this is the responsibility of the calling application.  For instance as described the EDN, docs.  (currently under review in #3391), that will be for us the most common attachment for this interface.  
Really what @mwbranstad has done here is created a full-featured interface to satisfy SP 800-90A, and then created the EDN to manage that according to our most common use-cases.

Reply:
In the typical configuration this will be managed by the EDN.

Reply:
Sorry, I was not explicit here - the text says "it is the responsibility of the application interface to reseed" - I was suggesting it's really the calling application that is responsible, not the application interface itself.

Reply:
Oh thanks for clarifying.  How does "...responsibility of the calling application..." sound?

Reply:
that's probably better and works for me.

Reply:
Also, I just realized that my previous comment was out of context... @mwbranstad is talking about tossing the `prediction_resistance_flag` out the window.  I'm not opposed to that I suppose, but we'll have to see what the lab says.